src/engine/execute.js

```diff
@@ -148,19 +140,19 @@ var execute = function (sequencer, thread) {
             execute(sequencer, thread);
             if (thread.status === Thread.STATUS_PROMISE_WAIT) {
                 return;
-            } else {
-                // Execution returned immediately,
-                // and presumably a value was reported, so pop the stack.
-                currentStackFrame.waitingReporter = null;
-                thread.popStack();
             }
+
+            // Execution returned immediately,
+            // and presumably a value was reported, so pop the stack.
+            currentStackFrame.waitingReporter = null;
+            thread.popStack();
         }
         argValues[inputName] = currentStackFrame.reported[inputName];
     }
 
     // Add any mutation to args (e.g., for procedures).
-    var mutation = blockContainer.getMutation(currentBlockId);
-    if (mutation) {
+    var mutation = blockContainer.getMutation(block);
+    if (mutation !== null) {
```

Comment:
I do profile everything - that's why I made the change. When I said it was peanuts in the grand scale of things I was comparing it with all the extra function calls, etc that occur continuously through the runtime. However, taken on there own, using `IF (a!==null)` is much faster than 'IF (a)'. At least under the conditions that I tested it ;).